sharktuner/sharktuner/libtuner.py

```diff
@@ -1292,6 +1305,17 @@ def benchmark(
         benchmark_time=benchmark_time,  # Only candidate benchmark has time limit.
     )
 
+    for res in candidate_results:
+        tuning_client.tuning_records[
+            res.candidate_id
+        ].benchmark_device_id = res.device_id
```

Comment:
why do we have to keep looking up the same candidate_id? Can we look it up and keep as a variable?